Support renv in R workflows

Felixmil opened the pull request.

Felixmil commented on .github/workflows/R-CMD-check-build.yaml:
```diff
@@ -35,24 +35,55 @@ jobs:
+              update = FALSE # avoid updating packages already installed by renv::restore()
+            )
+          }
+        shell: Rscript {0}
```
Installs necessary packages for next steps that may not be in `renv.lock`.
`update = FALSE` to avoid installing different versions than the ones that may already be specified in `renv.lock`

Yuri05 commented:
Let's add an optional workflow input `ignore-renv` which is **false** per default, but would allow skipping the whole renv setup if set to **true**.

Felixmil commented:
Update for @Yuri05: I realise the same renv/no renv logic needed to be implemented for test-coverage and pkgdown workflows. So I refactored the logic to a dedicated composite action and implemented this in other workflows.

Felixmil commented on .github/actions/setup-r-env/action.yml:
This is the reusable composite action

Felixmil commented on .github/workflows/R-CMD-check-build.yaml:
```diff
@@ -61,14 +76,15 @@ jobs:
-          echo "PKG_NAME=$(Rscript -e 'cat(pkgload::pkg_name())')" >>  $GITHUB_ENV
-          echo "PKG_VERSION=$(Rscript -e 'cat(as.character(desc::desc_get_version()))')" >> $GITHUB_ENV
-          echo "R_VERSION=$(Rscript -e 'cat(R.version$major, R.version$minor, sep = ".")')" >> $GITHUB_ENV
+          echo "PKG_NAME=$(Rscript --vanilla -e 'cat(pkgload::pkg_name())')" >>  $GITHUB_ENV
```
`--vanilla` is needed, otherwise, the R call returns `renv` loading console message

Felixmil commented on .github/workflows/R-CMD-check-build.yaml:
```diff
@@ -61,14 +76,15 @@ jobs:
-          echo "PKG_VERSION=$(Rscript -e 'cat(as.character(desc::desc_get_version()))')" >> $GITHUB_ENV
-          echo "R_VERSION=$(Rscript -e 'cat(R.version$major, R.version$minor, sep = ".")')" >> $GITHUB_ENV
+          echo "PKG_NAME=$(Rscript --vanilla -e 'cat(pkgload::pkg_name())')" >>  $GITHUB_ENV
+          echo "PKG_VERSION=$(Rscript --vanilla -e 'cat(as.character(desc::desc_get_version()))')" >> $GITHUB_ENV
```
> --vanilla
> Combine --no-save, --no-environ, --no-site-file, --no-init-file and --no-restore. Under Windows, this also includes --no-Rconsole.

So the package will be available and we don't care about renv being activated here.
